### DISCUSSION TITLE
Optional folder organization

### DISCUSSION THREAD

**Discussion by T-prog3** [TumblThreeApp/TumblThree#282] [Ideas]
**Is your feature request related to a problem? Please describe.**
TumblrThree becomes hard to keep organized over time. I have some blogs that is very large with 5000+ posts and most of their content is reblogged from other users. So when everything is in the same folder it's almost impossible to find the blogs unique content and look at it separately from the reblogged posts. At this point we no longer have a one blog per folder structure, it's more like 100 blogs in 1.

This leads me to a second problem. If i want to find out where any reblogged image is from i first have to make sure to have all metadata downloaded to get the `images.txt` file. Then i need to find that file, then do a filename search and finally copy the URL. This is such a tedious process just to get essential information if i want to download that blog.

**Describe the solution you'd like**
I would like to have optional settings to organize blogs content into smaller pieces to get a better overview. We already have the option to select what type of content to download. `Images, Videos, Audio, Text, Quote, Conversation, Link, Answer, Reblogged, Thumbnails, ImageMeta, VideoMeta, AudioMeta`. All of these could be grouped into separate folders or smaller category to make a distinction between different types of content and make it easier to find what you looking for. 
I would also like to have the option to separate reblogged content into their `reblog name:` folder. This could also help to identify new blogs you like and speed up selective deletes.

**It could be something like this:**
Lets say you have a section in settings that can activate preferred groups where you can choose what you would like to separate into a group. 

[  ] Images 
[  ] Video
[  ] Audio
[  ] Text files
[  ] Reblogged
-- [  ] Reblog name
[  ] Thumbnails

When the user have checked the box it will download that type of content into that folder. And if the box is unchecked it will download that type into the default blog folder. And if you want to combine some of the types you can then easily do it manually when everything else is out of the way.

**Comment by thomas694** [comment]
Thanks for your feature suggestion. We'll think about it.
Would this new setting be set for all blogs or individually for each blog? And I think sorting the files in subfolders based on reblog names could make it unclear/unmanageable too easily, so better to use the category only.

Two small hints how you can already solve it manually in a similar way. There aren't too many file extensions used, so the explorer list can be sorted by type or the search functionality can be used, and the respective group can be moved to a subfolder.
To be able to distinguish normal posts from reblogs and their origins you can use the filename template (e.g. `%o_%i_%f`). Then a filename from a reblog would be like `<reblog origin>_<post id>_<original filename>.ext` and normal ones start with `_<post id>`. And with the reblog name and the post id you can easily open that specific post in the browser without looking up anything.

**Comment by T-prog3** [comment]
My main thought is to allow the user to do their own preferred customization like what you already can do in other areas of the software. So the new setting would be a **default value** that can be changed on a blog to blog basis. 
It's all about the users own preferences like:

- Do i want thumbnails to be downloaded to their own category on this blog? 
- Do i want to keep all normal posts in the main folder but reblogs in a reblog category? 
- Do i want every type to be neatly organized into a category or just some? 
- Do i want **reblogs** in the reblogs folder to be organized in their `Reblog name` subfolder or everything in the same folder.

`Reblog name` as subfolder in the reblogs category is an especially needed option on **larger blogs** that can have over 200 reblogs from the same origin. In my workflow it would help with a better overview when blogs is scaled beyond 2000+ reblog posts, easier to identify blogs and delete the reblogs i don't like. It would also make me scroll less to find what I'm looking for.
